.changeset/fluffy-waves-ring.md

```diff
@@ -0,0 +1,5 @@
+---
+"stylelint": patch
+---
+
+Fixed: deprecation warnings to only display once per (custom) rule
```

Comment:
suggestion

```suggestion
Fixed: deprecation warnings to only display once per lint
```

Reply:
It is "once per unique warning each time Stylelint is loaded" but I am unsure how specific we have to be here :)

Maybe a bit more vague is actually better?

> Fixed: only display unique deprecation warnings

Reply:
Either vague or more specific, as you want.

Reply:
I think I actually prefer the original message :)
I think it best conveys what was changed to Stylelint users.

Reply:
The thing is we have deprecation warnings that are not related to rules.
e.g. `GITHUB_FORMATTER`
i.e. hence `per (custom) rule` is not really accurate IMO

Reply:
But from a user's perspective those warnings would not have changed.
Only the deprecations that were very repetitive have been reduced.

So even if the code change was more "global", the message is still informative and useful for Stylelint users.

But I also don't feel strongly about this :)
The original message seems more to the point to me.

I'll let @jeddy3 chime in, having described the original issue.